app/characterization_services/tika_file_characterization_service.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+# Class for Apache Tika based file characterization service
+# defines the Apache Tika based characterization service a ValkyrieFileCharacterization service
+# @since 0.1.0
+class TikaFileCharacterizationService
+  attr_reader :file_node, :persister
+  def initialize(file_node:, persister:)
+    @file_node = file_node
+    @persister = persister
+  end
+
+  # characterizes the file_node passed into this service
+  # Default options are:
+  #   save: true
+  # @param save [Boolean] should the persister save the file_node after Characterization
+  # @return [FileNode]
+  # @example characterize a file and persist the changes by default
+  #   Valkyrie::FileCharacterizationService.for(file_node, persister).characterize
+  # @example characterize a file and do not persist the changes
+  #   Valkyrie::FileCharacterizationService.for(file_node, persister).characterize(save: false)
+  def characterize(save: true)
+    result = JSON.parse(json_output).last
+    @file_characterization_attributes = { width: result['tiff:ImageWidth'], height: result['tiff:ImageLength'], mime_type: result['Content-Type'], checksum: checksum }
+    @file_node = @file_node.new(@file_characterization_attributes.to_h)
+    @persister.save(resource: @file_node) if save
+    @file_node
+  end
+
+  # Provides the SHA256 hexdigest string for the file
+  # @return String
+  def checksum
+    MultiChecksum.new(
```

Comment:
Looks like the process is very similar to how it works in Java: you read the file in chunks and update multiple checksum objects with each chunk: https://gist.github.com/escowles/acc2c0d5c5b3fc5724ae2e768c398d0b